starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,465 @@
 // Find the maximum
+var nb1 = 2;
+var nb2 = 3;
+
+function maxOfTwoNumbers(nb1, nb2) {
+  return Math.max(nb1, nb2);
+}
 
 // Finding Longest Word
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
 
+function findLongestWord(words) {
+  for (var i = 0; i < words.length; i++) {
+    var long = "";
+    if (words[i].length > long.length) {
+      long = words[i];
+    } else {
+      long = long;
+    }
+  }
+  return long1;
+}
+
+findLongestWord(words);
+
```

Comment:
You need to declare your variable inside the function but outside the `for loop` to store the value of the longest word.
The else is not necessary.
You are returning something different `long1` than the name you gave at the beginning of your variable `long` so it can't work.

Here is how you could have done it (but you were pretty close):
```
function findLongestWord(words) {
  var long = "";
  if (words.length === 0) {
    return undefined;
  }
  for (var i = 0; i < words.length; i++) {
    if (words[i].length > long.length) {
      long = words[i];
    } 
  }
  return long;
}

findLongestWord(words);
```